code/game/objects/items/dice.dm

```diff
@@ -428,16 +428,17 @@
 			human_servant.equipOutfit(/datum/outfit/butler)
 			var/datum/mind/servant_mind = new /datum/mind()
 			var/datum/antagonist/magic_servant/servant_antagonist = new
+			servant_mind.transfer_to(human_servant)
 			servant_mind.add_antag_datum(servant_antagonist)
```

Comment:
The antag's on_gain calls ``greet()``, why do we need to make it silent if they are already in their body when we're giving them the datum?

Reply:
> The antag's on_gain calls `greet()`, why do we need to make it silent if they are already in their body when we're giving them the datum?

it just didn't show when I tested it, no idea why. Giving antag manually did show the objectives, but specifically spawning through dice showed nothing. Putting a to_chat message anywhere before the ghost poll code also didn't show anything.

Reply:
Sorry for the long delay.

The problem is that they are creating a body, then giving it a mind, then giving the antag datum, then finally adding the person into the body.

The proper fix would be to poll ghosts for the servant and to put them in the body, THEN to give the antag datum, that way they properly get ``greet()`` and we won't need to snowflake this in case it's used more than just here.